### PR TITLE
created appexception

### LINKED ISSUE #68: create appexception class for bad requests

### PR DESCRIPTION
**Pull request by chrisK00**
close #68

### CODE REVIEW

File: Server/JokesOnYou.Web.Api/Exceptions/AppException.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
+
+namespace JokesOnYou.Web.Api.Exceptions
+{
+    public class AppException : Exception
+    {
+        public AppException() : base()
```
Comment (HellHunterMax) [inline Server/JokesOnYou.Web.Api/Exceptions/AppException.cs]: don't think the base here is needed ?
https://docs.microsoft.com/en-us/dotnet/standard/exceptions/how-to-create-user-defined-exceptions

**Review by HellHunterMax — APPROVED** [review]
Just wondering if the inner exception would be handy but we can always add it later if we need it.

**Comment by chrisK00** [comment]
> Just wondering if the inner exception would be handy but we can always add it later if we need it.

Its just hard for me think about. Like imagine a user field is null, we throw a KeyNotFoundException, our middleware catches it and sends back a 404 not found. When would we add more exceptions and how would that be sent back? To me it only makes sense in validation since your validating multiple stuff. If you have an example id love that.

File: Server/JokesOnYou.Web.Api/Exceptions/AppException.cs (hunk 1 above)
Comment (chrisK00) [reply Server/JokesOnYou.Web.Api/Exceptions/AppException.cs]: Correct. Its a matter of inheritance and usage of base keyword tho.

**Comment by HellHunterMax** [comment]
For example: the user register method returns with multiple errors inside an aggregate exception. You could add the aggregate exception as inner exception. Not sure if it's good but it's possible. 

I really don't know if we will need it. We will see when we get to that point and can always then add it.

**Review by code-reaper08 — APPROVED** [review]

**Pull request merged**